enderio-machines/src/main/java/com/enderio/machines/common/blockentity/VacuumChestBlockEntity.java

```diff
@@ -0,0 +1,132 @@
+package com.enderio.machines.common.blockentity;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.enderio.base.common.blockentity.sync.IntegerDataSlot;
+import com.enderio.base.common.blockentity.sync.SyncMode;
+import com.enderio.machines.common.MachineTier;
+import com.enderio.machines.common.blockentity.base.MachineBlockEntity;
+import com.enderio.machines.common.blockentity.data.sidecontrol.item.ItemHandlerMaster;
+import com.enderio.machines.common.blockentity.data.sidecontrol.item.ItemSlotLayout;
+import com.enderio.machines.common.menu.VacuumChestMenu;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.item.ItemEntity;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+
+public class VacuumChestBlockEntity extends MachineBlockEntity {
+	private static final double COLLISION_DISTANCE_SQ = 1 * 1;
+    private static final double SPEED = 0.025;
+    private static final double SPEED_4 = SPEED ;
+	private static final int MAX_RANGE = 6;
+	private int range = 6;
+
+	public VacuumChestBlockEntity(BlockEntityType<?> pType, BlockPos pWorldPosition,
+			BlockState pBlockState) {
+		super(MachineTier.STANDARD, pType, pWorldPosition, pBlockState);
+		add2WayDataSlot(new IntegerDataSlot(() -> this.range, this::setRange, SyncMode.GUI));
+	}
+
+	@Override
+	public AbstractContainerMenu createMenu(int containerId, Inventory inventory, Player player) {
+		return new VacuumChestMenu(this, inventory, containerId);
+	}
+	
+	@Override
+	public Optional<ItemSlotLayout> getSlotLayout() {
+		return Optional.of(ItemSlotLayout.basic(28,0));
+	}
+	
+	@Override
+	protected ItemHandlerMaster createItemHandler(ItemSlotLayout layout) {
+		return new ItemHandlerMaster(getIoConfig(), layout) {
+			@Override
+            protected void onContentsChanged(int slot) {
+                setChanged();
+            }
+			
+			@Override
+			public ItemStack insertItem(int slot, ItemStack stack, boolean simulate) {
+				if (slot == 27) {
+					return stack;
+				}
+				return super.insertItem(slot, stack, simulate);
+			}
+		};
+	}
+	
+	@Override
+	public void tick() {
+		super.tick();
+		if (this.getRedstoneControl().isActive(level.hasNeighborSignal(worldPosition))) {
+			this.collectItems(this.getLevel(), this.getBlockPos(), this.range);
+		}
+	}
+	
+	private void collectItems(Level level, BlockPos pos, int range) {
+		AABB area = new AABB(pos).inflate(range);
+		List<ItemEntity> itemEntities = level.getEntitiesOfClass(ItemEntity.class, area, e -> true); //TODO filter logic
```

Comment:
cache that result for a few ticks (maybe 5(?)) as this is a rather expensive operation. Also don't module just the global ticktime, but rather modulo the position hash or be hash, so that multiple vacuum chests don't search for item entities in the same tick, balancing that load over multiple ticks